test/src/sdk-harness/test_projects/tx_fields/mod.rs

```diff
@@ -300,26 +277,39 @@ mod tx {
     #[tokio::test]
     async fn can_get_inputs_count() {
         let (contract_instance, _, wallet, _) = get_contracts().await;
+        let message = &wallet.get_messages().await.unwrap()[0];
+
+        let mut builder = contract_instance.methods()
+            .get_tx_inputs_count()
+            .transaction_builder()
+            .await
+            .unwrap();
+        
+        wallet.adjust_for_fee(&mut builder, 1000).await.unwrap();
```

Comment:
You could refactor your `get_contracts` method to have both a message with and without data. The message without data could be used to pay for the transaction. So you would not have to `adjust_for_fee`

Reply:
Done.